### PR TITLE
add: gridCols in choiceMenu

### PR DESCRIPTION
**Pull request by shryics**

### CODE REVIEW

File: src/app/modules/choice-menu/choice-menu.component.ts [hunk 1]
```diff
@@ -7,11 +7,16 @@ export interface ActionDef {
   jumpToScene: string;
 }
 
+export interface ActionOptDef {
```
Comment (chigix) [inline src/app/modules/choice-menu/choice-menu.component.ts]: Why not put statement to implement this interface in the Component Class below?

File: src/app/modules/choice-menu/rdf-schema.d.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 interface ChoiceMenuActionLd extends ActionDef { }
+interface ChoiceMenuActionOptLd extends ActionOptDef { }
 
 export interface ChoiceMenuLd extends ComponentSchema {
   '@context': ['https://rengular.js.org/context/common.jsonld', {
     jumpToScene: { '@id': 'http://rengular.js.org/schema/nextScene', '@type': '@id' }
   }];
   '@type': 'ChoiceMenu';
   'ChoiceMenu:#choices': ChoiceMenuActionLd[];
+  'ChoiceMenu:#gridCols': ChoiceMenuActionOptLd[];
```
Comment (chigix) [inline src/app/modules/choice-menu/rdf-schema.d.ts]: Isn't this property defined as a number?

File: src/context/common.jsonld [hunk 3]
```diff
@@ -38,6 +38,7 @@
     "matchMedia": {"@id": "ren:matchMedia"},
     "css": {"@id": "ren:CssStyle#"},
     "backgroundImage": {"@id": "ren:backgroundImage", "@type": "@id"},
+    "gridCols" : {"@id":, "ren:gridCols, "@type": "@id},
```
Comment (chigix) [inline src/context/common.jsonld]: This is not needed. The schema should be defined and referenced as `http://rengular.js.org/schema/ChoiceMenu#gridCols`, which could be written in short as `ChoiceMenu:#gridCols` with the context `https://rengular.js.org/context/common.jsonld` in usage.

File: src/app/rengular-components-schema.ts [hunk 4]
```diff
@@ -83,6 +83,11 @@ const RENGULAR_REGISTRY: {
   'http://rengular.js.org/schema/ChoiceMenu': {
     component: ChoiceMenuComponent,
     inputs: {
+      'http://rengular.js.org/schema/ChoiceMenu#options': (component, data) => {
```
Comment (chigix) [inline src/app/rengular-components-schema.ts]: Function Style assignment is not needed here. 

    'http://rengular.js.org/schema/ChoiceMenu#gridCols' : gridCols

is all right.

With the definition above, special interface for the `gridCols`, `ActionOptDef` created here, could be removed.

File: src/app/modules/choice-menu/rdf-schema.d.ts [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
 interface ChoiceMenuActionLd extends ActionDef { }
+interface ChoiceMenuActionOptLd extends ActionOptDef { }
 
 export interface ChoiceMenuLd extends ComponentSchema {
   '@context': ['https://rengular.js.org/context/common.jsonld', {
     jumpToScene: { '@id': 'http://rengular.js.org/schema/nextScene', '@type': '@id' }
   }];
   '@type': 'ChoiceMenu';
   'ChoiceMenu:#choices': ChoiceMenuActionLd[];
+  'ChoiceMenu:#options': ChoiceMenuActionOptLd[];
```
Comment (chigix) [inline src/app/modules/choice-menu/rdf-schema.d.ts]: It should be 

```typescript
`ChoiceMenu:#gridCols`: number
```

File: src/app/modules/choice-menu/rdf-schema.d.ts [hunk 6]
```diff
@@ -1,12 +1,14 @@
 import { ComponentSchema, json } from 'app/renpi/rdf-schema';
-import { ActionDef } from './choice-menu.component';
+import { ActionDef, ActionOptDef } from './choice-menu.component';
 
 interface ChoiceMenuActionLd extends ActionDef { }
+interface ChoiceMenuActionOptLd extends ActionOptDef { }
```
Comment (chigix) [inline src/app/modules/choice-menu/rdf-schema.d.ts]: Basically, it seems not the actions of the menu or the buttons themselves, not like the buttons assigning covered by `ActionDef`, which may be could be considered that no additional interface definition needed here.

File: src/app/rengular-components-schema.ts [hunk 7]
```diff
@@ -83,6 +83,11 @@ const RENGULAR_REGISTRY: {
...
+        component.choices = data.map(tuple => ({
+          gridCols: tuple['http://rengular.js.org/schema/gridCols']['@id'],
+        } as ActionOptDef));
+      },
```
Comment (chigix) [inline src/app/rengular-components-schema.ts]: Follow your definition, the data json would become:

```json
{
  "@type": "http://rengular.js.org/schema/ChoiceMenu",
  "http://rengular.js.org/schema/ChoiceMenu#choices": [...buttons...],
  "http://rengular.js.org/schema/ChoiceMenu#options": [
    {
      "@type": "....The type of the gridCols ...",
      "http://rengular.js.org/schema/gridCols": {
        "@id": 2
      }
    }
  ]
}
```

Then why not make it concise without decreasing the intuition:

```json
{
  "@type": "http://rengular.js.org/schema/ChoiceMenu",
  "http://rengular.js.org/schema/ChoiceMenu#choices": [...buttons...],
  "http://rengular.js.org/schema/gridCols": 2,
}
```

File: src/app/modules/choice-menu/choice-menu.component.ts [hunk 8]
```diff
@@ -12,6 +12,11 @@ export interface ActionDef {
   templateUrl: './choice-menu.component.html',
   styleUrls: ['./choice-menu.component.scss']
 })
+
+export interface ActionOptDef {
+  gridCols: number;
```
Comment (chigix) [inline src/app/modules/choice-menu/choice-menu.component.ts]: Upon the comments in the files below, there seems no need to create additional interface for the defined properties on the `ChoiceMenuComponent` itself.

**Comment by chigix** [comment]
> please rename this commit as fix: gridCols schema miss in ChoiceMenu

@shryics , pay attention to the commit message writing.

File: src/app/rengular-components-schema.ts (hunk 4 above)
Comment (shryics) [reply src/app/rengular-components-schema.ts]: Following your advice, ```gridCols``` is undefined.

Comment (chigix) [reply src/app/rengular-components-schema.ts]: Sorry, quotes are missing: 

'http://rengular.js.org/schema/ChoiceMenu#gridCols' : 'gridCols'

Such practice is used in other components as well:

https://github.com/chigix/rengular/blob/de329617500e112895f7c9de684fe47814677f0f/src/app/rengular-components-schema.ts#L60

Comment (shryics) [reply src/app/rengular-components-schema.ts]: I get it. Thank you.

File: src/app/modules/choice-menu/rdf-schema.d.ts [hunk 9]
```diff
@@ -1,14 +1,13 @@
...
-interface ChoiceMenuActionOptLd extends ActionOptDef { }
 
 export interface ChoiceMenuLd extends ComponentSchema {
   '@context': ['https://rengular.js.org/context/common.jsonld', {
     jumpToScene: { '@id': 'http://rengular.js.org/schema/nextScene', '@type': '@id' }
   }];
   '@type': 'ChoiceMenu';
   'ChoiceMenu:#choices': ChoiceMenuActionLd[];
-  'ChoiceMenu:#options': ChoiceMenuActionOptLd[];
+  'ChoiceMenu:#options': number
```
Comment (chigix) [inline src/app/modules/choice-menu/rdf-schema.d.ts]: Should it be `ChoiceMenu:#gridCols` ?

File: src/app/modules/choice-menu/choice-menu.component.ts [hunk 10]
```diff
@@ -12,6 +12,7 @@ export interface ActionDef {
   templateUrl: './choice-menu.component.html',
   styleUrls: ['./choice-menu.component.scss']
 })
+
```
Comment (chigix) [inline src/app/modules/choice-menu/choice-menu.component.ts]: Remove no-content change.

**Pull request merged**